src/backend/cdb/motion/ic_udpifc.c

```diff
@@ -1167,6 +1169,105 @@ resetRxThreadError()
 }
 
 
+/*
+ * isIPv6Available
+ * 		Check whether a valid IPv6 address exists in the system.
+ *
+ * 		We want to have both loopback IPv6 address, which will show if IPv6 was
+ * 		disabled earlier, and some non-link-local address currently available
+ * 		for our use.
+ */
+static bool
+isIPv6Available(void)
+{
+	bool		hasIpv6Addr = false;
+	bool		hasIpv6Loopback = false;
+	int			errnoSave = errno;
+
+	int			errorErrnoSave = 0;
+	char	   *errorCause = NULL;
+
+	struct ifaddrs	*ifaddr = NULL;
+	struct ifaddrs	*ifaddrFirst = NULL;
+
+	if (getifaddrs(&ifaddrFirst) == -1)
+	{
+		errorCause = "getifaddrs()";
+		goto end;
+	}
+
+	for (ifaddr = ifaddrFirst; ifaddr != NULL; ifaddr = ifaddr->ifa_next)
```

Comment:
```suggestion
	for (struct ifaddrs *ifaddr = ifaddrFirst; ifaddr != NULL; ifaddr = ifaddr->ifa_next)
```

Reply:
That's mixing declarations and code. Every other function in this file declares it's variables at the top of the block, and I followed that style.